crates/router/src/connector/paypal/transformers.rs

```diff
@@ -853,3 +865,101 @@ pub struct PaypalAccessTokenErrorResponse {
     pub error: String,
     pub error_description: String,
 }
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct PaypalWebhooksBody {
+    pub event_type: PaypalWebhookEventType,
+    pub resource: PaypalResource,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+pub enum PaypalWebhookEventType {
+    #[serde(rename = "PAYMENT.AUTHORIZATION.CREATED")]
+    PaymentAuthorizationCreated,
+    #[serde(rename = "PAYMENT.AUTHORIZATION.VOIDED")]
+    PaymentAuthorizationVoided,
+    #[serde(rename = "PAYMENT.CAPTURE.DECLINED")]
+    PaymentCaptureDeclined,
+    #[serde(rename = "PAYMENT.CAPTURE.COMPLETED")]
+    PaymentCaptureCompleted,
+    #[serde(rename = "PAYMENT.CAPTURE.PENDING")]
+    PaymentCapturePending,
+    #[serde(rename = "PAYMENT.CAPTURE.REFUNDED")]
+    PaymentCaptureRefunded,
+    #[serde(rename = "CHECKOUT.ORDER.APPROVED")]
+    CheckoutOrderApproved,
+    #[serde(rename = "CHECKOUT.ORDER.COMPLETED")]
+    CheckoutOrderCompleted,
+    #[serde(rename = "CHECKOUT.ORDER.PROCESSED")]
+    CheckoutOrderProcessed,
+    #[serde(other)]
+    Unknown,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+#[serde(untagged)]
+pub enum PaypalResource {
+    PaypalCardWebhooks(Box<PaypalCardWebhooks>),
+    PaypalRedirectsWebhooks(Box<PaypalRedirectsWebhooks>),
+    PaypalRefundWebhooks(Box<PaypalRefundWebhooks>),
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct PaypalRefundWebhooks {
+    pub id: String,
+    pub amount: OrderAmount,
+    pub seller_payable_breakdown: PaypalSellerPayableBreakdown,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct PaypalSellerPayableBreakdown {
+    pub total_refunded_amount: OrderAmount,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct PaypalCardWebhooks {
+    pub supplementary_data: PaypalSupplementaryData,
+    pub amount: OrderAmount,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct PaypalRedirectsWebhooks {
+    pub purchase_units: Vec<PaypalWebhooksPurchaseUnits>,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct PaypalWebhooksPurchaseUnits {
+    pub reference_id: String,
+    pub amount: OrderAmount,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct PaypalSupplementaryData {
+    pub related_ids: PaypalRelatedIds,
+}
+#[derive(Deserialize, Debug, Serialize)]
+pub struct PaypalRelatedIds {
+    pub order_id: String,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct PaypalWebooksEventType {
+    pub event_type: PaypalWebhookEventType,
+}
+
+impl From<PaypalWebhookEventType> for api::IncomingWebhookEvent {
+    fn from(event: PaypalWebhookEventType) -> Self {
+        match event {
+            PaypalWebhookEventType::PaymentCaptureCompleted
+            | PaypalWebhookEventType::PaymentAuthorizationCreated
```

Comment:
Why are we marking `PaymentAuthorizationCreated` and `PaymentAuthorizationVoided` as PaymentIntentSuccess?

Reply:
This events are moved to event not supported